go/lang/security/audit/xss/template-html-does-not-escape.yaml

```diff
@@ -0,0 +1,20 @@
+rules:
+- id: template-html-does-not-escape
+  message: |
+    Semgrep could not determine that the argument to 'template.HTML()'
+    is a constant. 'template.HTML()' does not escape contents.
+    Be absolutely sure there is no user-controlled data in this 
+    template. If user data can reach this template, you may have
+    a XSS vulnerability.
+  metadata:
+    cwe: "CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
+    owasp: 'A1: Injection'
+    references:
+    - https://golang.org/pkg/html/template/#HTML
+    - https://github.com/0c34/govwa/blob/139693e56406b5684d2a6ae22c0af90717e149b8/vulnerability/xss/xss.go#L33
+  languages: [go]
+  severity: WARNING
+  patterns:
+  - pattern-not: template.HTML("..." + "...")
```

Comment:
could there be an addition of more than 2 constants?

Reply:
Yeah, but this stemmed from a platform job which revealed a handful of repositories doing this. Basically this is just here to reduce FP rate by a small amount.